### ISSUE TITLE
Access to remote Payara seems broken after v11.3

### ISSUE THREAD

**Issue by AndrewG10i** [apache/netbeans#7804]
### Apache NetBeans version

Apache NetBeans 23

### What happened

I had to work with a remote Payara server (remote domain) running on remote Linux machine. I am able to successfully access and login to DAS via browser, but when adding remote Payara domain to NetBeans 23 - NetBeans doesn't show it running.

I have tried several versions of NetBeans, and seems the last NetBeans which was able to access remote Payara domain and show it online - is NetBeans 11.3. After that version - no versions are able to show remote Payara domain as online.

<details>
  <summary>Click to expand screenshot</summary>

  ![nb-11 3-works](https://github.com/user-attachments/assets/a1dedbbb-13df-46f7-8beb-4357f8a4f8b1)
  ![nb-12 6-error](https://github.com/user-attachments/assets/5cc241f9-6532-49da-9571-6fa010a61136)

</details>

### Language / Project Type / NetBeans Component

Payara Server integration

### How to reproduce

1) Download latest Payara 6 Community Edition
2) Launch its domain1 as standalone server (commands to execute: ./asadmin change-admin-password, ./asadmin start-domain domain1, ./asadmin enable-secure-admin), and make sure you can connect to its DAS via https://server.name.com:4848/ using login and password
3) Download latest NetBeans 23
4) Go to the Servers tab and add new Payara Server as follows:
- Select: Remote Domain
- Domain name: domain1
- Host name: DNS-name of the machine where Payara Server domain1 launched
- Ports use default
- Login and Password use as defined on the step 2
5) Once server added - try to open/refresh it
6) ISSUE: Added remote Payara Server has to be shown with green play arrow, indicating that it is running, BUT this doesn't happen and server is showed as off-line.

### Did this work correctly in an earlier version?

Apache NetBeans 16 or earlier

### Operating System

Windows 10

### JDK

Azul Zulu 17

### Apache NetBeans packaging

Apache NetBeans binary zip

### Are you willing to submit a pull request?

No

**Comment by matthiasblaesing** [comment]
I tested without secure-admin and that works as it should and that might make sense. I'm missing the step where you install a valid TLS certificate for the Payara Server. Did you just don't mention that or did you not?

Please provide the message.log of a session where you tested the access and the contents is missing.

**Comment by AndrewG10i** [comment]
Hi! Thanks for the reply. Could you pls clarify what you mean "tested without secure-admin"? In case you mean you have added "Local Domain" - yes, that works fine.

The issue I have reported is about trying to add "Remote Domain", pls. see the screenshot below:
![image](https://github.com/user-attachments/assets/8736290d-ae2c-4c1d-ae14-b51e72969f77)

In order for NB to be able to control (deploy) app on the Remote Domain - DAS has to be activated on the remote Payara (at least it was so ~6 years ago when I last used such approach ;) ).

As per certificates - no, I was not installing any additional certificates, as it has to work by default with the existing certificate coming with Payara/Glassfish.

Messages.log - attaching. I have closed NB23, deleted previous message.log, launched NB23, added Remote Domain, tried to access it / refresh it, nothing worked. Closed NB23 and copied `messages.log`.
[messages.log](https://github.com/user-attachments/files/17179989/messages.log)

Let me know pls in case further clarifications are required from me.
Thank you!

**Comment by matthiasblaesing** [comment]
I tested in remote mode and retestet. Please check if right clicking on the server entry and choosing "Refresh" resolves it.

**Comment by AndrewG10i** [comment]
Can you pls list your steps? Also could you pls clarify what you mean "tested without secure-admin" as per your first reply?
"Refresh" - of course it was tried as well, didn't help.
Thanks!

**Comment by matthiasblaesing** [comment]
Ok - my suspicion was, that the remote control code in the payara extension might fail without a valid certificate (I think it should), but I tested with MDNS resolver (`<hostname>.local`) and connection worked with and without secure-admin (aka with http and https on port 4848).

For the steps I followed your requested steps.

I tried to find the right loggers to get the right debug output and this should do it (run with cmd):

```
<path_to_netbeans>/bin/netbeans64.exe -J-Dorg.netbeans.modules.payara.common.CommonServerSupport.level=FINEST -J-Dorg.netbeans.modules.payara.tooling.server.state.StatusJob.level=FINEST -J-Dorg.netbeans.modules.payara.tooling.server.state.StatusJob$ListenerPortCheck.level=FINEST -J-Dorg.netbeans.modules.payara.tooling.level=FINEST
```

**_DO NOT POST THE RESULT WITHOUT REVIEW!!! The URL, user and password for the DAS will be printed!_**

I see this (password replaced by `PPPPPPPPP`, hostname replaced by `HHHHHHHHHHHH`):

```
FINER [org.netbeans.modules.payara.tooling.server.state.RunnerTask]: Asadmin command task for server Payara Server in state ONLINE started.
FINE [org.netbeans.modules.payara.tooling.utils.ServerUtils]: ServerUtils.isSecurePort.socket
FINEST [org.netbeans.modules.payara.tooling.admin.Runner]: Runner call() method called on URL http://HHHHHHHHHHHH:4848/__asadmin/version thread Payara Status Tasks
FINEST [org.netbeans.modules.payara.tooling.admin.Runner]: Setting HTTP connection attributes: timeout = 3.000 user = admin password = PPPPPPPPP
FINEST [org.netbeans.modules.payara.tooling.admin.Runner]: Using "Basic" authorization with user and password
FINEST [org.netbeans.modules.payara.tooling.admin.Runner]: Connecting to HHHHHHHHHHHH:4848
FINE [org.netbeans.modules.payara.tooling.admin.Runner]: Receiving response, code: 302
FINE [org.netbeans.modules.payara.tooling.admin.Runner]: Location moved to https://HHHHHHHHHHHH:4848/__asadmin/version
FINEST [org.netbeans.modules.payara.tooling.admin.Runner]: Setting HTTP connection attributes: timeout = 3.000 user = admin password = PPPPPPPPP
FINEST [org.netbeans.modules.payara.tooling.admin.Runner]: Using "Basic" authorization with user and password
FINEST [org.netbeans.modules.payara.tooling.admin.Runner]: Connecting to HHHHHHHHHHHH:4848
FINE [org.netbeans.modules.payara.tooling.admin.Runner]: Receiving response, code: 200
FINEST [org.netbeans.modules.payara.tooling]: Reading response from HHHHHHHHHHHH:4848
FINEST [org.netbeans.modules.payara.tooling]: Manifest message: Payara Server 6.2024.9 #badassfish (build 23)
FINE [org.netbeans.modules.payara.tooling.server.state.StatusJob$ListenerVersion]: Status check version command result: SUCCESS
FINE [org.netbeans.modules.payara.tooling.server.state.StatusJob]: Command check transition SUCCESS: ONLINE -> ONLINE
````

This makes sense. The status check connects to the DAS, gets a redirect from http to https and the queries the `/__asadmin/version` path to query the server version.

What do you get?

**Comment by AndrewG10i** [comment]
I am getting the following result:
```
FINER [org.netbeans.modules.payara.tooling.server.state.RunnerTask]: Asadmin command task for server Payara Server in state OFFLINE_PORT started.
FINE [org.netbeans.modules.payara.tooling.utils.ServerUtils]: ServerUtils.isSecurePort.socket
FINEST [org.netbeans.modules.payara.tooling.admin.Runner]: Runner call() method called on URL http://srv1.local:4848/__asadmin/version thread Payara Status Tasks
FINEST [org.netbeans.modules.payara.tooling.admin.Runner]: Setting HTTP connection attributes: timeout = 3,000 user = admin password = PPPP
FINEST [org.netbeans.modules.payara.tooling.admin.Runner]: Using "Basic" authorization with user and password
FINEST [org.netbeans.modules.payara.tooling.admin.Runner]: Connecting to srv1.local:4848
FINE [org.netbeans.modules.payara.tooling.admin.Runner]: Receiving response, code: 302
FINE [org.netbeans.modules.payara.tooling.admin.Runner]: Location moved to https://srv1.local:4848/__asadmin/version
FINEST [org.netbeans.modules.payara.tooling.admin.Runner]: Setting HTTP connection attributes: timeout = 3,000 user = admin password = PPPP
FINEST [org.netbeans.modules.payara.tooling.admin.Runner]: Using "Basic" authorization with user and password
FINEST [org.netbeans.modules.payara.tooling.admin.Runner]: Connecting to srv1.local:4848
FINE [org.netbeans.modules.payara.tooling.admin.Runner]: Receiving response, code: 200
FINEST [org.netbeans.modules.payara.tooling]: Reading response from srv1.local:4848
FINEST [org.netbeans.modules.payara.tooling]: Manifest message: Payara Server 6.2024.9 #badassfish (build 23)
FINE [org.netbeans.modules.payara.tooling.server.state.StatusJob$ListenerVersion]: Status check version command result: FAILED
FINE [org.netbeans.modules.payara.tooling.server.state.StatusJob]: Command check transition FAILED: OFFLINE_PORT -> OFFLINE_PORT
```

Difference with your output is the following: `Status check version command result: FAILED`. I am running NB23 on Win10 with Azul Java 17.
Thanks!

**Comment by matthiasblaesing** [comment]
I can reproduce it. What is the version of the payara installation you are using to register the remote domain?

**Comment by AndrewG10i** [comment]
Thank you for your reply and I'm glad that you were able to reproduce the issue!

As per versions - initially I was testing it as follows:
- on local machine I had version 6.2024.6 (but I was sure that it was 6.2024.5 - thus I got the issue);
- on remote machine I tried versions 6.2024.5 and 6.2024.9 - that didn't work as reported;

Having your input in mind - I aligned payara versions on my local machine and remote machine:
- once I aligned local version and remote to have v6.2024.5 on both ends - I was able to see remote Payara online status properly:
![image](https://github.com/user-attachments/assets/19d4699c-2889-4efb-a840-5115b3391737)

Thank you for helping to resolve this issue! I do remember that before (~6 years ago when I last used deployment to remote domain) it was not making any difference if remote and local versions were bit different (as my test on NB 11.3 showns), thus I didn't pay much attention to it.
In case it is a requirement now to have the same versions of payara on local and remote machines - I am fine with that and absolutely okay to have this issue closed.
From the other side if, at least, some warnings about local vs. remote versions mismatch could be added in scope of this issue - it would be beneficial for other users of NB who potentially could get in the same situation.
Thus it is up to you how to proceed with this issue.

Thank you once again! ;)
